pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/StagePlanner.java

```diff
@@ -104,6 +112,16 @@ private StageNode walkRelPlan(RelNode node, int currentStageId) {
     }
   }
 
+  // Could run CBO here later
+  private void runPhysicalOptimizers(QueryPlan queryPlan) {
```

Comment:
Should we wrap PhysicalOptimizer in a separate class if there is a lot more extension in the future?

Reply:
for now it is fine. let's consider the SPI for worker assignment and we can refactor later